classes/tdx-optee.bbclass

```diff
@@ -0,0 +1,40 @@
+# enable OP-TEE support
+TDX_OPTEE_ENABLE = "1"
+
+# required by some vendor BSPs
+MACHINE_FEATURES:append = " optee"
+
+# enable OP-TEE debug messages
+TDX_OPTEE_DEBUG ?= "0"
+
+# enable installation of OP-TEE test applications
+TDX_OPTEE_INSTALL_TESTS ?= "0"
+
+# OP-TEE test applications
+TDX_OPTEE_PACKAGES_TESTS = "\
+    optee-test \
+    optee-examples \
+"
+
+# extra packages for OP-TEE support
+IMAGE_INSTALL:append = "\
+    optee-os \
+    optee-client \
+    ${@oe.utils.conditional('TDX_OPTEE_INSTALL_TESTS', '1', '${TDX_OPTEE_PACKAGES_TESTS}', '', d)} \
+"
+
+# validate optee support
+addhandler validate_optee_support
+validate_optee_support[eventmask] = "bb.event.SanityCheck"
+python validate_optee_support() {
+    supported_machines = ['verdin-imx8mp']
+    machine = e.data.getVar('MACHINE')
+    if machine not in supported_machines:
+        bb.fatal("OP-TEE is currently not supported on '%s' machine!" % machine)
+
+    if 'tdx-signed' in d.getVar('OVERRIDES').split(':'):
+        bb.warn("Secure boot might not work because the OP-TEE firmware is not being signed!")
```

Comment:
Could you elaborate more on the possible consequences here?

When you say "might not work", is it not deterministic what would happen here?

Reply:
Hmmm. Good catch. Wrong wording indeed. I changed to "Secure boot will not work because the OP-TEE firmware is currently not being signed!". PR updated.

Reply:
Thanks for the clarification. So then, if this will not work at all is `bb.warn` still appropriate here or would it make more sense to be `bb.fatal`?

Reply:
Allright, so I thought about it. Although I am already working on a fix for this, indeed it makes more sense for now to just fail with an error if both secure boot and OP-TEE are enabled. PR updated.